AMP_CACHE_GUIDELINES.md

```diff
@@ -0,0 +1,85 @@
+# AMP Cache Guidelines
+
+## Purpose
+
+Provide guidelines to external parties for implementing their own AMP Caches.
+
+## Guidelines: Core
+
+1. An AMP Cache only serves [valid AMP documents](https://github.com/ampproject/amphtml/blob/master/spec/amp-html-format.md).
+
+2. It participates in the AMP Project's [validator release cycle](https://github.com/ampproject/amphtml/tree/master/validator).
+
+3. It participates in the AMP JS library release cycle and makes every effort to serve the latest version. It does not allow sites to perform version locking.
+
+4. Pledges to maintain URL space forever (even beyond the lifetime of the cache itself): 
+
+    1. This can be achieved by donating the URL space to a trustworthy third party entity such as [archive.org](http://archive.org).
+
+    2. This means that, should a cache decide to no longer operate, URLs should redirect to the origin URL or be served by another cache.
+
+5. Uses HTTPS serving only.
+
+6. Serves a strong [Content Security Policy](https://w3c.github.io/webappsec-csp/) ([CSP](https://developer.chrome.com/extensions/contentSecurityPolicy)) - a CSP defines a mechanism by which web developers can control the resources which a particular page can fetch or execute, as well as a number of security-relevant policy decisions. The Content-Security-Policy meta-tag reduces the risk of XSS attacks by defining where resources can be loaded from, thus preventing browsers from fetching data via potentially risky locations:
+
+    3. Implements separately documented security rewrites on served content.
+
+    4. The CSP should be equivalent or stronger to what is being served on [cdn.ampproject.org](http://cdn.ampproject.org).
+
+    5. Caches must update their CSP in a timely fashion (within 7 days) at the request of the AMP Project.
+
+7. Supports a public Update ping mechanism: 
+
+    6. Equivalent to the [Google AMP Cache Update ping format](https://developers.google.com/amp/cache/update-ping#update-ping-format)
```

Comment:
Although it would be nice to mandate a single URL format for the API for ease of implementation on the publisher side.